packages/blockchain-wallet-v4-frontend/src/data/modules/profile/sagas.ts

```diff
@@ -29,10 +29,22 @@ export const renewUserDelay = 30000
 let renewSessionTask = null
 let renewUserTask = null
 export default ({ api, coreSagas, networks }) => {
-  const waitForUserData = function * () {
-    const userId = (yield select(
+  const waitForUserId = function * () {
+    const userId = yield select(
+      selectors.core.kvStore.userCredentials.getUserId
+    )
+    if (Remote.Success.is(userId)) return userId.getOrElse(null)
+    yield take(
+      actionTypes.core.kvStore.userCredentials
+        .FETCH_METADATA_USER_CREDENTIALS_SUCCESS
```

Comment:
If user credentials metadata fails to load (`FETCH_METADATA_USER_CREDENTIALS_FAILURE `) then this take will never resolve.

Reply:
@plondon should we trow an error in case of failure? I think it make sense

Reply:
actually never mind will apply same approach we do in waitForUserData